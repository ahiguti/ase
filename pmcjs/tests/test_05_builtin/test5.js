
/* SKIP_CHECKMEM (fork() causes confusion) */

var core = Perl.Functions['CORE'];

core.system('echo abc');
core.system('echo', 'abcdef');

var pid = core.fork();
if (pid == 0) {
  print('child1');
  core.exit(3);
}
var k = core.waitpid(pid, 0);
var st = Perl.CHILD_ERROR >> 8;
print(st);

var pid = core.fork();
if (pid == 0) {
  core.exec('echo', 'child2');
}
var k = core.wait();
var st = Perl.CHILD_ERROR >> 8;
print(st);

