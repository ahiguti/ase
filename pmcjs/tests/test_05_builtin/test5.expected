abc
abcdef
child1
child2
3
0
